dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/util/TextUtils.java

```diff
@@ -68,6 +68,18 @@
 
     private static final String OPTION_SEP = ";";
 
+    /**
+     * Remove all non-alphanumeric characters within string
+     *
+     * @param str input string
+     * @return string with only alphanumeric characters and spaces, dash and
+     *         underscore
+     */
+    public static String removeNonEssentialChars( String str )
```

Comment:
Perhaps name this method something a bit more descriptive? Like retainAlphaNumeric (not fully accurate) or escapeLogStatement, or similar.